layers.go

```diff
@@ -491,7 +491,10 @@ func (r *layerStore) saveMounts() error {
 	return r.loadMounts()
 }
 
-func (s *store) newLayerStore(rundir string, layerdir string, driver drivers.Driver) (LayerStore, error) {
+// newLayerStore creates a new layer storage based on the provided directories
+// and selected driver. The `withMountsLock` bool indicates if we want to
```

Comment:
Is it really `withMountsLock` or `readWrite`? The latter seems semantically more useful, and might e.g. allow skipping the `MkdirAll` calls below.